### DISCUSSION TITLE
raw.githubusercontent.com does not update

### DISCUSSION THREAD

**Discussion by GladistonXD** [community/community#46758] [Repositories]
### Select Topic Area

Bug

### Body

**I can't update old files with the same name.**

**Comment by 2Abendsegler** [comment]
Hi,

I can confirm that. I have the same problem. 

In the meantime I have deleted my two main branches "master" and "collector" in my repository. Nevertheless, the raw.githubusercontent.com links are still there. This is only the case with these two special branches. With other branches, the file under the raw.githubusercontent.com link is also immediately deleted when deleting the branch. I can create the two branches again from another one, but the raw.githubusercontent.com are not updated. 

Some of the raw.githubusercontent.com links are large scripts (1.2 MB). This may have something to do with the problem, because it always takes a while before the merge of a branch is also visible in the raw.githubusercontent.com link.

To Reproduce:
- Go to repository: https://github.com/2Abendsegler/GClh
You can see there is only one branch "master2".
- Call raw.githubusercontent.com link to a script in deleted branch "master": https://raw.githubusercontent.com/2Abendsegler/GClh/master/gc_little_helper_II.user.js
- And call raw.githubusercontent.com link to a script in deleted branch "collector": https://raw.githubusercontent.com/2Abendsegler/GClh/collector/gc_little_helper_II.user.js

Thanks for helping.

**Comment by cyroxx** [comment]
Same here.

The file https://github.com/robert-koch-institut/SARS-CoV-2-Nowcasting_und_-R-Schaetzung/blob/main/Nowcast_R_aktuell.csv was updated by this commit: https://github.com/robert-koch-institut/SARS-CoV-2-Nowcasting_und_-R-Schaetzung/commit/c9fae5633495603fed2dda3b5f235279d00fabb7

But the raw file still contains the old version, even though the commit is more than 6 hours old now:
https://raw.githubusercontent.com/robert-koch-institut/SARS-CoV-2-Nowcasting_und_-R-Schaetzung/main/Nowcast_R_aktuell.csv

**Reply by 2Abendsegler** [reply]
And it is only a small file. :(
In the meantime I [reported a bug](https://support.github.com/contact/bug-report) in the support area. However, [the ticket for this](https://support.github.com/ticket/personal/0/2005506) is probably not visible to everyone. There are no results yet.

**Reply by cyroxx** [reply]
I also filed a bug report, ticket #2008051.

**Comment by copyminter** [comment]
We are seeing the same issue with our files: https://github.com/teia-community/teia-report/blob/main/restricted.json
An update 3 hours ago is still not showing up in the raw version. I attempted to update the list to see if that triggers anything but it is still stuck on an old version from yesterday.

**Reply by copyminter** [reply]
This list that was updated 6 hours ago is now up to date with raw access. Another list I updated later still has the older updates.

**Reply by copyminter** [reply]
Another update 11 hours ago is still not showing up in the raw version. It used to take 5-10 mins for any updates to be reflected via raw.

**Reply by 2Abendsegler** [reply]
> Another update 11 hours ago is still not showing up in the raw version. It used to take 5-10 mins for any updates to be reflected via raw.

That can not be. New CEO Nat Friedman said after the acquisition that he wanted to make GitHub better for developers. 😂

**Comment by FlorianWilhelm** [comment]
Hi, I can confirm this bug. For me it's the following two files:

- https://github.com/FlorianWilhelm/pytanis/blob/main/docs/assets/images/logo.svg
- https://raw.githubusercontent.com/FlorianWilhelm/pytanis/main/docs/assets/images/logo.svg

Which should show the same logo but the latter one still shows the old version.

**Comment by alams154** [comment]
Also confirming the bug:

* [XML File](https://github.com/alams154/Hama.bundle/blob/main/offsets.xml)
* [XML File (raw)](https://raw.githubusercontent.com/alams154/Hama.bundle/main/offsets.xml)

File was changed about a hour ago and still has not updated the version available at raw.githubusercontent.com

**Comment by zezzy63** [comment]
Also confirming the bug:
https://github.com/zezzy63/fluffy-waddle/blob/main/Unrevealed.LoaderHash.txt
https://raw.githubusercontent.com/zezzy63/fluffy-waddle/main/Unrevealed.LoaderHash.txt

**Comment by Plextora** [comment]
I'm also confirming this bug as well. All files I try to update in the [Imrglop/Latite-Releases repo](https://github.com/Imrglop/Latite-Releases) do not update. For example, a commit that updates the [injector file](https://github.com/Imrglop/Latite-Releases/blob/main/injector/Injector.exe) does not go into effect on raw.githubusercontent.com

**Comment by niklas389** [comment]
Same here:
https://github.com/niklas389/lemonkebab_Nameless_A13/blob/master/update.json
https://raw.githubusercontent.com/niklas389/lemonkebab_Nameless_A13/master/update.json

**Comment by LeonmDK3** [comment]
Guess it might have something to do with this? "cache-control | max-age=86400"
Not sure but that would mean to cache the file for 24 hours instead of 600 sec which would be 5 minutes. 
Also affecting my repo: https://github.com/LeonmDK3/EmergencyAssistSystem/blob/main/unames.txt

**Reply by GladistonXD** [reply]
That could be right default repositories apparently is "max-age=300" repositories with a delay of up to 24h show 86400 in response

**Reply by real-yfprojects** [reply]
I don't know why but wget gets the right version, while my firefox still displays the old one although no-cache is enabled in the dev options.

**Comment by wiegerthefarmer** [comment]
Confirming I have the same bug. I made a typo in the one yaml file that our system pulls from raw.... and now my jupyter hub is down. Lol. Sigh.

**Reply by wiegerthefarmer** [reply]
CDN is now delivering the correct version.

**Comment by BangDroid** [comment]
Pulling my freakin hair out over here (what's left of it!). Usually when you have a thought like, GitHub is "broken", you are wrong and the problem is you. But it's nice (worse?) to know that occasionally, GitHub is indeed at fault.

**Reply by 2Abendsegler** [reply]
Off topic:
> Pulling my freakin hair out over here (what's left of it!). 

😂

> But it's nice (worse?) to know that occasionally, GitHub is indeed at fault.

That's not decided yet. 
Maybe because of the ever-increasing traffic, that's just the way it's intended. That's called deceleration. So like in the Stone Age, when you had to ask whether an e-mail had already arrived.

**Reply by BangDroid** [reply]
I'm not the one using the term _bug_, see above. All I meant is that I'm glad it's not just me, because 9/10 it is.

**Reply by wiegerthefarmer** [reply]
Right? I was working on my laptop tethered to my phone driving yesterday. I thought about it all night. Come into work early today to continue the work and then find others having the same issue. I feel much better.

**Reply by wiegerthefarmer** [reply]
Well exactly 24 hours later, my changes went through.

**Comment by Juanescacha** [comment]
I don't know if I'm right but I faced the same issue with an image from my repo on the Repo Readme, I changed it and didn't update after like 48h so I think its just a matter of time to update, and github is not constantly updating the changes to the urls, you will have to wait some time...

**Comment by GladistonXD** [comment]
The only way that "resolve" was changing the name of the repository, I will have to redirect the updates to the new name and after 24 hours change the name of the repository. 😑

**Reply by auwsom** [reply]
Could you clarify.. so change repo name works, but only after 24hrs (assuming because of the cache mag-age delay above), and you will have to redirect to the new name, but why "after 24 hours, change the name of the repository"? Why change it after 24 hours? Do you mean, change the name in the redirection after waiting 24hrs? (no, that wouldnt help.. idk)
Thx

**Reply by GladistonXD** [reply]
Changing the name of the repository resolves it immediately, but I have to wait 24 hours to redirect the repository updates to the new name, this is a specific case of mine, but changing the name of the repository apparently resolves it immediately.

**Reply by auwsom** [reply]
thanks, below I comment that copying the file to a new name also works.. have not tested replacing the original one with it.. but certainly less dramatic in complexity than changing a repo name if it is connected to other things. cheers.

**Comment by karsiong** [comment]
lolz.....anyone tag GitHub staff here let them know the issue, I think they didn't realize the issue after so many days.

**Reply by 2Abendsegler** [reply]
I have a free account. That probably applies to most people here. Our opinions probably don't carry much weight. It would certainly help if paying users made themselves known to GitHub, if they are affected by the problems at all.

**Comment by auwsom** [comment]
This is very frustrating. I use the raw content to download basic settings for Linux dotfiles. Has been working for years.. now broken.

https://stackoverflow.com/questions/53491680/curl-and-wget-not-getting-the-latest-committed-file-from-github

For some reason the 'raw' tab on GH website and URL shows the correct info.. but wget and curl do not retrieve it. How can this be?

Also weird, I can make a copy,  git commit, add and push it,  and the new file curls nearly instantly and correctly. (mystified)

**Comment by auwsom** [comment]
posting a new comment if anyone above needs a quick fix.. making a copy of the file might work.. might even be able to delete the non-curling one and then rename the working one to the original name (havent tested).

**Reply by Plextora** [reply]
Making a copy does indeed work, but for people using raw files for auto updaters (like me), this is not a good solution. Completely ruins the entire idea of auto updating.

**Reply by GladistonXD** [reply]
@Plextora My case is the same, I gave an example above of a possible solution redirecting the updates to a new repository, I don't see any other solution.

**Comment by 2Abendsegler** [comment]
If you create a branch with a previously unused name from another, the raw data of the files are up-to-date immediately, even if the raw data in the other branch was not up-to-date. 

Deleting a branch and then renaming the newly created one to the previously deleted one doesn't work. The original raw data of the deleted branch is preserved and is not immediately up to date.

So, besides changing the repository name, this could also be a possible solution to bypass the time lag. But automatic updaters are also problematic here.

**Comment by MichaelTiemannOSC** [comment]
Add my name to the list of people both experiencing this problem and also incredulous that there has been no response from GitHub.  The file not working for me is https://raw.githubusercontent.com/os-climate/rmi-utility-transition-hub-ingestion-pipeline/iceberg/dbt/rmi_transform/target/manifest.json (which I updated 6 hours ago).

**Comment by kristofferR** [comment]
A GitHub engineer (@antn) posted in the other thread: :)

`👋 Hi folks, GitHub engineer here. This was a bug with our caching in some cases - sorry about this! We're working on resolving this and the cache should be back to normal soon.`

https://github.com/orgs/community/discussions/46691#discussioncomment-4948308

**Reply by khengyun** [reply]
> A GitHub engineer (@antn) posted in the other thread: :)
> 
> `👋 Hi folks, GitHub engineer here. This was a bug with our caching in some cases - sorry about this! We're working on resolving this and the cache should be back to normal soon.`
> 
> https://github.com/orgs/community/discussions/46691#discussioncomment-4948308

thanks for the reply, but this error seems to have been around for about 2 weeks now. Last week I had the same problem like this, but it was updated after 10 minutes to 20 minutes. and now my file hasn't been updated all day

**Reply by weisez** [reply]
glad to hear, thought for a minute that i broke my script bruh 💀

**Comment by khengyun** [comment]
> ### Select Topic Area
> Bug
> 
> ### Body
> **I can't update old files with the same name.**

i have same problem here
https://github.com/khengyun/khengyun/blob/main/devcard.svg
and 
https://raw.githubusercontent.com/khengyun/khengyun/main/devcard.svg

**Comment by breslavsky** [comment]
Can confirm that have the same problem with

https://raw.githubusercontent.com/open-tutorials/cypress/main/assets/build_conduit/github_fork.webp

https://github.com/open-tutorials/cypress/blob/main/assets/build_conduit/github_fork.webp

**Reply by 2Abendsegler** [reply]
Look here https://github.com/orgs/community/discussions/46691#discussioncomment-4948987

**Comment by ItachiPvPUchiha** [comment]
Same here:

https://github.com/ItachiPvPUchiha/CuHub/blob/main/yba.lua

https://raw.githubusercontent.com/ItachiPvPUchiha/CuHub/main/yba.lua

Fix it please.

**Comment by GladistonXD** [comment]
I did some tests the problem is now in the default encoding, `Accept-Encoding: gzip` , if you use `Accept-Encoding: deflate` it updates to the new version.

**Reply by heqds** [reply]
How do I use that encoding? I am having this problem 3 weeks later.

**Reply by GladistonXD** [reply]
These are the encodings of a request, you can send a get with headers on any platform.

`curl -sH 'Accept-encoding: deflate' http://example.com/...`

**Reply by heqds** [reply]
Where do I put that?

**Reply by GladistonXD** [reply]
On any request, take a look at this thread there may be better support. #46691

**Comment by kristoftorok** [comment]
I have the same problem with DNS block lists.

**Comment by Mandark-droid** [comment]
+1 getting same issue while trying to download. https://raw.githubusercontent.com/iterativv/NostalgiaForInfinity/main/NostalgiaForInfinityX2.py

Please fix.

**Reply by antn** [reply]
We are not setting a longer cache value for raw files. You're probably seeing the default cache value of 5 minutes. Please wait for the cache to be cleared.

**Comment by auragit** [comment]
Adding `?token=$(date +%s)` to the end of the URL worked for me.

**Reply by MichaelTiemannOSC** [reply]
Wow!  Somebody knows how to read the docs closely!!

**Comment by FarisFarley** [comment]
Now its working..

I just updated my repository from desktop git and it updated immediately but in (raw-github-user-content) it updated after 10 minutes.
Just note that when ever you update also change the description before you commit and when you push select the latest description then select the file you made changes and then push and it worked fine.

**Comment by avivash** [comment]
Same issue over here:

https://github.com/oddsdk/blockpaperscissors/blob/main/hamb.fundring

https://raw.githubusercontent.com/oddsdk/blockpaperscissors/main/hamb.fundring

**Comment by Harshkesharwani** [comment]
https://raw.githubusercontent.com/nltk/nltk_data/gh-pages/packages/tokenizers/punkt.zip
I could not download this please give me the solution

**Reply by MichaelTiemannOSC** [reply]
See https://github.com/nltk/nltk_data/issues/202.  Also see issues files in nltk/nltk that reference same problem.

**Comment by Xiaod0ng** [comment]
I made a minor update to my json file, but the raw file is not updated until 10 mins later.

**Comment by farishajdarpasic** [comment]
Same issue. I pushed, but raw is not updated

**Comment by kleinlennart** [comment]
Sad that this hasn't been fixed...
I'm using raw.githubusercontent.com to include JS scripts, and this bug is completely ruining my workflow.

**Comment by MTDickens** [comment]
Same issue here. Why don't Github team fix this? Or at least give us some replies...

But adding `?token=$(date +%s)` to the end of the URL worked for me.

**Reply by gchartier** [reply]
I think this doesn't work anymore as it was working for me and now I just get 404's

**Comment by koder101** [comment]
I can confirm that the issue still persists.

**Comment by Map987** [comment]
edit a text file , and the taw textfile isn't change real time .

**Comment by marionzr** [comment]
This is how I solved it yesterday.

To force GitHub to give you the latest version of a file without caching, you can append a query parameter to the URL. This query parameter ensures that GitHub treats the request as a new one, bypassing any cached version.

Example

```
https://raw.githubusercontent.com/marionzr/file-share/refs/heads/main/content.csv?timestamp=20250212
```

The timestamp=20250212 (or any unique value) ensures the URL is seen as different every time you request it, so GitHub will fetch the latest version of the file. So, alternatively, you can use `?cache_bust=<random_value>` to achieve the same effect. 

```
https://raw.githubusercontent.com/marionzr/file-share/refs/heads/main/content.csv?cache_bust=banana123
```

If that solves the problem for you, consider giving a look into my projects and giving them a star!

**Reply by sagarchaulagai** [reply]
its not working for me

**Reply by noa-santo** [reply]
it worked once and then not anymore after trying again later so it might have been a coincide the first time

**Comment by suzuki-shunsuke** [comment]
We faced the similar issue.

- https://github.com/aquaproj/aqua/issues/4068

We created a support ticket and GitHub is intestigating the issue.
As a workaround, we could get the correct file by specifying full length commit SHA.
